libs/core/thread_support/src/set_thread_name.cpp

```diff
@@ -83,16 +81,33 @@ namespace hpx::util {
         }
     }    // namespace detail
 
-    // Set the name of the thread shown in the Visual Studio debugger
-    void set_thread_name(char const* thread_name) noexcept
+    void set_thread_name(const char* thread_name) noexcept
     {
         detail::set_thread_name(thread_name);
+    }
+}    // namespace hpx::util
+
+#elif defined(HPX_HAVE_PTHREAD_SETNAME_NP)
 
-        // also set it the 'new' way in case the application is not running in
-        // the debugger at this time
-        SetThreadDescription(
-            GetCurrentThread(), detail::to_wide_string(thread_name).c_str());
+#include <pthread.h>
+namespace hpx::util {
+
+    void set_thread_name(char const* thread_name) noexcept
+    {
+        pthread_set_name_np(pthread_self(), thread_name);
```

Comment:
The feature test looks for `pthread_setname_np`, not `pthread_set_name_np`

Reply:
We seem to have missed this fact, but freeBSD and openBSD use ` pthread_set_name_np` while the remaining use pthread_setname_np. I will make changes accordingly to the feature test and `set_thread_name.cpp`

Reply:
Tests are a bit misleading as we don't seem to be testing for BSD.

Reply:
Testing for `pthread_set_name_np` should be separate and independent. Let's finalize the current use case (testing for `pthread_setname_np`) first.